core/src/main/java/org/elasticsearch/common/settings/loader/JsonSettingsLoader.java

```diff
@@ -27,6 +27,10 @@
  */
 public class JsonSettingsLoader extends XContentSettingsLoader {
 
+    public JsonSettingsLoader(boolean guardAgainstNullValuedSettings) {
```

Comment:
is there any place we want this to be false? just wondering if we just make it default behavior...